src/main/scala/xiangshan/mem/lsqueue/VirtualLoadQueue.scala

```diff
@@ -247,31 +242,8 @@ class VirtualLoadQueue(implicit p: Parameters) extends XSModule
       val need_rep = io.ldin(i).bits.rep_info.need_rep
       val need_valid = io.ldin(i).bits.updateAddrValid
 
-      when (!need_rep && need_valid) {
-      // update control flag
-        addrvalid(loadWbIndex) := hasExceptions || !io.ldin(i).bits.tlbMiss || io.ldin(i).bits.isSWPrefetch
-        datavalid(loadWbIndex) :=
-          (if (EnableFastForward) {
-              hasExceptions ||
-              io.ldin(i).bits.mmio ||
-             !io.ldin(i).bits.miss && // dcache miss
-             !io.ldin(i).bits.dcacheRequireReplay || // do not writeback if that inst will be resend from rs
-              io.ldin(i).bits.isSWPrefetch
-           } else {
-              hasExceptions ||
-              io.ldin(i).bits.mmio ||
-             !io.ldin(i).bits.miss ||
-              io.ldin(i).bits.isSWPrefetch
-           })
-
-        //
-        when (io.ldin(i).bits.data_wen_dup(1)) {
-          uop(loadWbIndex) := io.ldin(i).bits.uop
-        }
-        when (io.ldin(i).bits.data_wen_dup(4)) {
-          uop(loadWbIndex).debugInfo := io.ldin(i).bits.uop.debugInfo
-        }
-        uop(loadWbIndex).debugInfo := io.ldin(i).bits.rep_info.debug
+      when (!need_rep && need_valid && !isvec(loadWbIndex)) {
```

Comment:
Would it be better to use `!io.ldin(i).bits.isvec` here? It would avoid introducing three read ports. :kissing_closed_eyes: